### PR TITLE
Enable usage of S3 as a target for media upload & serving

### PR DESCRIPTION
**Pull request by lmorchard**
- Add django-storages and settings to use S3 for media
- Split up docker-compose so local customizations are easier

### CODE REVIEW

**Comment by lmorchard** [comment]
Still wrapping my head around our deployment strategy with elasticbeanstalk. But, using Amazon S3 as storage and web hosting for uploaded media (e.g. images for experiments) seems like a good idea.

Even if we don't end up using S3, django-storages seems handy for [all the backends it offers](https://github.com/jschneier/django-storages/tree/master/storages/backends) besides the basic local filesystem.

**Comment by jaredhirsch** [comment]
I can take a look at this

File: docker-compose-s3.yml-dist [hunk 1]
```diff
@@ -0,0 +1,18 @@
+db:
+  image: postgres:9.3
+client_build:
+  extends:
+    file: docker-compose-base.yml
```
Comment (jaredhirsch) [inline docker-compose-s3.yml-dist:5]: Aha, so this 'extends' thing gives us a really simple way to toggle between prod and dev, yes? We just modify the env vars?

**Comment by jaredhirsch** [comment]
Got this working without any problems.

Might make sense to also serve static files (styles, scripts) from the same S3 bucket; that'd make CDNification really simple. But it sounds like scope creep for the current issue, so I'll file a bug to discuss (#141).

Merging.

File: docker-compose-s3.yml-dist (hunk 1 above)
Comment (lmorchard) [inline docker-compose-s3.yml-dist:5]: Yeah, I'm hoping we can keep the image for the Django server built in a way that the only dev / prod difference is env vars. Don't think we'll be able to use docker-compose in prod, but we should be able to play with prod-ish features in dev like S3 by just juggling env vars

**Pull request merged**